core/src/test/java/com/yandex/yoctodb/util/mutable/impl/FoldedByteArrayIndexedListTest.java

```diff
@@ -0,0 +1,261 @@
+package com.yandex.yoctodb.util.mutable.impl;
+
+import com.yandex.yoctodb.util.UnsignedByteArray;
+import com.yandex.yoctodb.util.UnsignedByteArrays;
+import com.yandex.yoctodb.util.buf.Buffer;
+import com.yandex.yoctodb.util.mutable.ByteArrayIndexedList;
+import org.junit.Test;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.LinkedHashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+import static com.yandex.yoctodb.util.UnsignedByteArrays.from;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+
+public class FoldedByteArrayIndexedListTest {
+
+    @Test
+    public void testTwoBytesFolding() {
+        testManyValues(1000);
+    }
+
+    @Test
+    public void testFourBytesFolding() {
+        testManyValues(70000);
```

Comment:
This test is extremely slow -- it takes almost half a minute on my laptop. Is that really needed?